lib/driver.js

```diff
@@ -762,8 +762,7 @@ class XCUITestDriver extends BaseDriver {
       // local helper for the two places we need to uninstall wda and re-start it
       const quitAndUninstall = async (msg) => {
         this.log.debug(msg);
-        const noUninstallCap = ['webDriverAgentUrl', 'usePreinstalledWDA']
-          .find((x) => Boolean(this.opts[x]));
+        const noUninstallCap = !this.isXcodebuildNeeded();
```

Comment:
I would say we don't need this intermediate variable here. Simply move `!this.isXcodebuildNeeded()` directly to the if condition